electron; Build recipe for Electron 37.1.0 OS-19689

caneraltinbasak opened the pull request:
Added an electron build recipe and supplementary files. Electron recipes include Chromium recipes and override the necessary parts like do_compile, do_configure.

caneraltinbasak committed:
electron; Build recipe for Electron 37.1.0

Added an electron build recipe and supplementary files. Electron
recipes include Chromium recipes and override the necessary parts
like do_compile, do_configure.

t-bashir-bs commented on meta-chromium/recipes-browser/chromium/electron.inc:
```diff
@@ -0,0 +1,42 @@
+PV = "37.1.0-bs.alpha2"
+
+# Electron-specific source - replace Chromium source with Electron source
+SRC_URI = "https://electron-ci-public-artifact-bucket.s3.amazonaws.com/v${PV}/v${PV}.tar.xz"
```
I assume that Electron don't make tarballs, and we would need that if we want to upstream this and not have it take customised Brightsign releases?

caneraltinbasak replied:
There is no official Electron tarball. We will compile&test all our changes. I don't think it would be a problem for anyone.

t-bashir-bs commented on meta-chromium/recipes-browser/chromium/electron.inc:
```diff
@@ -0,0 +1,42 @@
+PV = "37.1.0-bs.alpha2"
+
+# Electron-specific source - replace Chromium source with Electron source
+SRC_URI = "https://electron-ci-public-artifact-bucket.s3.amazonaws.com/v${PV}/v${PV}.tar.xz"
```
I assume the nice "LOCAL BUILD" option we had before has been removed as you wanted a clearer version to upstream later?

caneraltinbasak replied:
I'll do these types of things in bbappend to keep it separate from upstream.

caneraltinbasak committed:
electron,chromium: Add automation for building Electron

meta-chromium-test is updated to test electron, as well as chromium.
Added an electron workflow to use that infrastructure.

caneraltinbasak committed:
electron-ozone-x11: Add a recipe for ozone-x11 version of Electron

caneraltinbasak committed:
fix(electron): remove aarch64 compile fix

Electron already seems to have this patch applied.

caneraltinbasak committed:
Build fix

caneraltinbasak committed:
github-action: Test X11 builds as well as Wayland

t-bashir-bs commented on meta-chromium/recipes-browser/chromium/electron-ozone-wayland.bb:
```diff
@@ -0,0 +1,46 @@
+        use_system_libwayland=false \
+        use_system_minigbm=true \
+        use_system_libffi=true \
+        use_bundled_fontconfig=false \
```
Should we make this an override, as this is a specific change we made for our system?
